sdk/src/main/java/com/google/cloud/dataflow/sdk/options/GcpOptions.java

```diff
@@ -183,24 +184,67 @@ void setCredentialFactoryClass(
   public static class DefaultProjectFactory implements DefaultValueFactory<String> {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProjectFactory.class);
 
+    private static String getUserHome() {
+      return System.getProperty("user.home");
+    }
+
+    private String getActiveConfig() throws IOException {
+      final String defaultConfigName = "default";
+
+      // get active configuration
+      File activeConfFile = new File(
+          getUserHome(),
+          ".config/gcloud/active_config");
```

Comment:
Sorry @lukecwik - was on vacation. I see, thanks for the gcloud-java link, I looked at the code, and it's pretty much the same. afaiu current DF does not check for:

```
    String projectId = System.getProperty(PROJECT_ENV_NAME, System.getenv(PROJECT_ENV_NAME));
    if (projectId == null) {
      projectId = appEngineProjectId();
    }
    if (projectId == null) {
      projectId = serviceAccountProjectId();
    }
```

Specifically `serviceAccountProjectId` could be interesting, afaiu it would discover if user is using a service account through `GOOGLE_APPLICATION_CREDENTIALS` and then if available, fetch the project from the credentials file. One could argue if that is good or bad idea, probably good for `defaultProject`.

It seems like we could push this PR, I could submit PR to gcloud-java to abstract the project discover logic away from `ServiceOptions`, then I could submit another PR to DF to start using gcloud-java? Does it sound like a good plan?